spack fails to build patch@2.7.5 because of symlink problems

Hi all,

yesterday I did a fresh spack install on a virtual machine with a minimal OS installation (CentOS 7 with only few packages, including Python 2.7, git, curl and GCC 4.8.5) that still fulfills the minimal requirement for spack.

The first thing that I noticed is, that I need to build patch, but this already failed:

```
[sfux@develop02 ~]$ spack purge -a
[sfux@develop02 ~]$ spack find
==> 0 installed packages.
[sfux@develop02 ~]$ spack install patch
==> Installing patch
==> Fetching http://ftp.gnu.org/gnu/patch/patch-2.7.5.tar.xz
######################################################################## 100.0%
==> Staging archive: /local/cluster/spack/var/spack/stage/patch-2.7.5-3jgpfxlxg35dn4l4duohkkoombxag4oo/patch-2.7.5.tar.xz
==> Created stage in /local/cluster/spack/var/spack/stage/patch-2.7.5-3jgpfxlxg35dn4l4duohkkoombxag4oo
==> Ran patch() for patch
==> Building patch [AutotoolsPackage]
==> Executing phase : 'autoreconf'
==> Executing phase : 'configure'
==> Executing phase : 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j8'
/local/cluster/spack/lib/spack/spack/build_systems/autotools.py:275, in build:
     270      def build(self, spec, prefix):
     271          """Makes the build targets specified by
     272          :py:attr:``~.AutotoolsPackage.build_targets``
     273          """
     274          with working_dir(self.build_directory):
  >> 275              inspect.getmodule(self).make(*self.build_targets)

See build log for details:
  /local/scratch/sfux/spack-stage/spack-stage-DFYEr8/patch-2.7.5/spack-build.out
```

Looking at the error message:

```
make: GNUmakefile: Too many levels of symbolic links
make: stat: GNUmakefile: Too many levels of symbolic links
make: *** No rule to make target `GNUmakefile'.  Stop.
==> Error: ProcessError: Command exited with status 2:
```

it seems that there is a symlink pointing to itself:

lrwxrwxrwx 1 sfux sfux-group    105 Mar  9 09:55 GNUmakefile -> /local/cluster/spack/var/spack/stage/patch-2.7.5-3jgpfxlxg35dn4l4duohkkoombxag4oo/patch-2.7.5/GNUmakefile


Any ideas ?

Please let me know what additional information I can provide to resolve this issue.

Best regards

Sam

@alalazo I was running the commands that you suggested:

```
[sfux@develop02 cluster]$ git clone https://github.com/llnl/spack.git 
Cloning into 'spack'...
remote: Counting objects: 57041, done.
remote: Compressing objects: 100% (49/49), done.
remote: Total 57041 (delta 21), reused 3 (delta 3), pack-reused 56988
Receiving objects: 100% (57041/57041), 20.47 MiB | 7.62 MiB/s, done.
Resolving deltas: 100% (27448/27448), done.
[sfux@develop02 cluster]$ cd spack/
[sfux@develop02 spack]$ git fetch origin pull/3338/head:master
remote: Counting objects: 7, done.
remote: Compressing objects: 100% (3/3), done.
remote: Total 7 (delta 0), reused 0 (delta 0), pack-reused 4
Unpacking objects: 100% (7/7), done.
From https://github.com/llnl/spack
 * [new ref]         refs/pull/3338/head -> master
[sfux@develop02 spack]$ git checkout master
Switched to branch 'master'
[sfux@develop02 spack]$
```

But I still get the same error message about the symlinks:

```
[sfux@develop02 ~]$ spack install patch
==> Installing patch
==> Fetching http://ftp.gnu.org/gnu/patch/patch-2.7.5.tar.xz
######################################################################## 100.0%
==> Staging archive: /local/cluster/spack/var/spack/stage/patch-2.7.5-3jgpfxlxg35dn4l4duohkkoombxag4oo/patch-2.7.5.tar.xz
==> Created stage in /local/cluster/spack/var/spack/stage/patch-2.7.5-3jgpfxlxg35dn4l4duohkkoombxag4oo
==> Ran patch() for patch
==> Building patch [AutotoolsPackage]
==> Executing phase : 'autoreconf'
==> Executing phase : 'configure'
==> Executing phase : 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j8'
/local/cluster/spack/lib/spack/spack/build_systems/autotools.py:275, in build:
     270      def build(self, spec, prefix):
     271          """Makes the build targets specified by
     272          :py:attr:``~.AutotoolsPackage.build_targets``
     273          """
     274          with working_dir(self.build_directory):
  >> 275              inspect.getmodule(self).make(*self.build_targets)

See build log for details:
  /local/scratch/sfux/spack-stage/spack-stage-GxZLiT/patch-2.7.5/spack-build.out
```

and

```
config.status: linking /local/cluster/spack/var/spack/stage/patch-2.7.5-3jgpfxlxg35dn4l4duohkkoombxag4oo/patch-2.7.5/GNUmakefile to GNUmakefile
config.status: executing depfiles commands
==> 'make' '-j8'
make: GNUmakefile: Too many levels of symbolic links
make: stat: GNUmakefile: Too many levels of symbolic links
make: *** No rule to make target `GNUmakefile'.  Stop.
```

Best regards

Sam

--- a/var/spack/repos/builtin/packages/patch/package.py
+++ b/var/spack/repos/builtin/packages/patch/package.py
@@ -35,3 +35,5 @@ class Patch(AutotoolsPackage):
     url      = "http://ftp.gnu.org/gnu/patch/patch-2.7.5.tar.xz"
 
     version('2.7.5', 'e3da7940431633fb65a01b91d3b7a27a')
+
+    build_directory = 'spack-build'

